### PR TITLE
fix data inconsistent bug in tests

### PR DESCRIPTION
**Pull request by zsluedem**
## Overview
Fix the tests data inconsistent bug

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: comm/src/test/scala/coop/rchain/p2p/EffectsTestInstances.scala [hunk 1]
```diff
@@ -63,26 +64,36 @@ object EffectsTestInstances {
     type Responses = PeerNode => Protocol => CommErr[Unit]
     var reqresp: Option[Responses] = None
     var requests: List[Request]    = List.empty[Request]
+    val lock                       = new SyncVar[TransportLayerStub[F]]()
+    lock.put(this)
```
Comment (tgrospic) [inline comm/src/test/scala/coop/rchain/p2p/EffectsTestInstances.scala]: There is `SyncVarOps.create` helper function to initialize SyncVar.

File: casper/src/test/scala/coop/rchain/casper/engine/InitializingSpec.scala [hunk 2]
```diff
@@ -116,10 +116,8 @@ class InitializingSpec extends WordSpec with BeforeAndAfterEach {
         // Assert requested messages for the state and fork choice tip
         _        = assert(transportLayer.requests.size == expectedRequests.size)
         messages = transportLayer.requests.map(_.msg)
-        _ = messages.zip(expectedRequests).map {
-          case (msg1, msg2) => assert(msg1 == msg2)
-        }
-        _ = transportLayer.reset()
+        _        = assert(messages.forall(m => expectedRequests.exists(e => e == m)))
+        _        = transportLayer.reset()
```
Comment (tgrospic) [inline casper/src/test/scala/coop/rchain/casper/engine/InitializingSpec.scala:120]: This works but it's not precise enough. We should expect all requests correctly and not just part of it.

Order can be non-deterministic so comparing sets can be enough.
`assert(messages.toSet == expectedRequests.toSet)`

Maybe also delay in line 102 of 1 second is not enough to create the same requests. :thinking:

Comment (zsluedem) [reply casper/src/test/scala/coop/rchain/casper/engine/InitializingSpec.scala:120]: `forall` is not enough?

Comment (tgrospic) [reply casper/src/test/scala/coop/rchain/casper/engine/InitializingSpec.scala:120]: Not sure what you mean.

Comment (tgrospic) [reply casper/src/test/scala/coop/rchain/casper/engine/InitializingSpec.scala:120]: P.S. Overall, I'm not sure what this test is supposed to check. If just Initializing engine why all other stuff need to be created, like everything (block store, dag store, casper, ...) and also in the same way as in normal node operation, with LMDB and on file system.

Comment (tgrospic) [reply casper/src/test/scala/coop/rchain/casper/engine/InitializingSpec.scala:120]: I changed delay to 5 seconds. This should be enough for now.
https://github.com/rchain/rchain/pull/3134/files#diff-91178b6525f827b57ac86c3d7c1a3cd1R102

**Review by tgrospic — APPROVED** [review]
LGTM

**Review by nzpr — APPROVED** [review]
LGTM

**Pull request merged**